### PR TITLE
feat(css): add resource CSS es core upgrade

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
add resource CSS es core upgrade

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```
add resource CSS es core upgrade
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/css" TESTARGS="-run TestAccCssEsCoreUpgrade_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run TestAccCssEsCoreUpgrade_basic -timeout 360m -parallel 4
=== RUN   TestAccCssEsCoreUpgrade_basic
=== PAUSE TestAccCssEsCoreUpgrade_basic
=== CONT  TestAccCssEsCoreUpgrade_basic
--- PASS: TestAccCssEsCoreUpgrade_basic (2261.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       2261.368s
```

### CODE REVIEW

File: huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+		Pending:      []string{"RUNNING"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      upgradeCoreStateRefreshFunc(client, clusterId, id),
+		Timeout:      timeout,
+		PollInterval: 10 * timeout,
+		Delay:        10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for CSS (%s) to be extend: %s", clusterId, err)
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go]: this msg seems not suitable

File: huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+	if id == "" {
+		return diag.Errorf("not found task ID of the upgrade core")
+	}
+	d.SetId(id)
+
+	// Check whether upgrade core has completed.
+	err = checkUpgradeCoreCompleted(ctx, client, clusterID, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		err1 := retryEsUpgradeCore(client, d)
+		if err != nil {
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go]:
```suggestion
		if err1 != nil {
```

File: docs/resources/css_es_upgrade_core.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+---
+subcategory: "Cloud Search Service (CSS)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_css_es_upgrade_core"
+description: |-
+  Manages CSS ElasticSearch upgrade core resource within HuaweiCloud.
+---
+
+# huaweicloud_css_es_upgrade_core
```
Comment (Jason-Zhang9309) [inline docs/resources/css_es_upgrade_core.md]:
```suggestion
# huaweicloud_css_es_core_upgrade
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+	id := utils.PathSearch("id", upgradeDetail, "").(string)
+	if id == "" {
+		return diag.Errorf("not found task ID of the upgrade core")
+	}
+	d.SetId(id)
+
+	// Check whether upgrade core has completed.
+	err = checkUpgradeCoreCompleted(ctx, client, clusterID, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		goBackErr := removeEsUpgradeCoreImpact(client, d)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go]: better to replace `go back` with `roll back`
```suggestion
		rollBackErr := removeEsUpgradeCoreImpact(client, d)
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_core_upgrade.go [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+
+	expression := "detailList | [?status=='RUNNING'] | [0]"
+	upgradeDetail, err := getEsUpgradeDetail(client, clusterID, expression)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := utils.PathSearch("id", upgradeDetail, "").(string)
+	if id == "" {
+		return diag.Errorf("not found task ID of the upgrade core")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_core_upgrade.go]: don't forget the error messages and comments
```suggestion
		return diag.Errorf("not found task ID of the core upgrade")
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_core_upgrade.go [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+	}
+	d.SetId(id)
+
+	// Check whether upgrade core has completed.
+	err = checkCoreUpgradeCompleted(ctx, client, clusterID, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		rollBackErr := removeEsCoreUpgradeImpact(client, d)
+		if rollBackErr != nil {
+			return diag.Errorf("error upgrade core: %v, going back upgrade fail: %s", err, rollBackErr)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_core_upgrade.go]:
```suggestion
			return diag.Errorf("error upgrade core: %v, rolling back upgrade fail: %s", err, rollBackErr)
```

**Pull request merged**